Parking-Zone/Services/ReservationService.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Parking_Zone.Models;
+using Parking_Zone.Repositories;
+
+namespace Parking_Zone.Services
+{
+    public class ReservationService : Service<Reservation>, IReservationService
+    {
+        public ReservationService(IReservationRepository repository) : base(repository)
+        {
+        }
+
+        public override void Insert(Reservation entity)
+        {
+            entity.Id = Guid.NewGuid();
```

Comment:
Move this line to base Service class (do the same for other inherited services as well).